test/unit/org/apache/cassandra/db/KeyCacheTest.java

```diff
@@ -338,6 +340,24 @@ private void testKeyCache(String cf) throws ExecutionException, InterruptedExcep
         assertKeyCacheSize(noEarlyOpen ? 4 : 2, KEYSPACE1, cf);
     }
 
+    @Test
+    public void testKeyCacheLoadZeroCacheLoadTimeBackwardCompatibility() throws Exception
+    {
+        // Pre-4.1 users were asked to use negatives to disable. This test validates the backward compatibility
+        // in case a user still uses their old yaml format and negative values for this property.
+        DatabaseDescriptor.setCacheLoadTimeout(-1);
```

Comment:
...but doesn't `setCacheLoadTimeout()` here coerce the value to zero and, in fact, not test `NEGATIVE_SECONDS_DURATION` w/ `cache_load_timeout_seconds`? Seems like we'd want to reject negative values in `setCacheLoadTimeout()` but test w/ something like `cassandra-old.yaml` where `cache_load_timeout_seconds: -1` right?

Reply:
Maybe like a special little version of `cassandra-old.yaml` that just looks at old default/disabled values in `LoadOldYAMLBackwardCompatibilityTest`?

Reply:
Yes, I am writing better `Converters`'s tests to cover both loading from yaml and using `fromMap` method as part of another ticket. I will share it soon.